### ISSUE TITLE
Parameters seems not being updated

### ISSUE THREAD

**Issue by hanyangclarence** [om-ai-lab/VLM-R1#291]
Thanks for your great work!

I am new to RL. I am working on the grpo_jsonl.py training scripts, and tried to print the mean values of the parameters on each training step, but I found that the values are not changed.  I print the model's parameter values as:

```python
def _generate_and_score_completions(...):
  # ...
  mean_values = 0.0
  for name, param in model.base_model.named_parameters():
      if param.requires_grad:
          mean_values += param.abs().mean()
  if self.accelerator.is_main_process:
      print(f"!! Mean absolute value of trainable parameters: {mean_values.item(): .8f}")
  # ...
```

And the output is like:
```
!! Mean absolute value of trainable parameters:  116.50000000
!! Mean absolute value of trainable parameters:  116.50000000
{'loss': 0.0, 'grad_norm': 2.3918297290802, 'learning_rate': 9.999751033212169e-07, 'completion_length': 99.5546875, 'rewards/iou_reward': 0.07502234354615211, 'rewards/format_reward_rec': 0.0859375, 'reward': 0.1609598472714424, 'reward_std': 0.369953915476799, 'kl': 0.00087738037109375, 'clip_ratio': 0.0, 'epoch': 0.0}
  0%|                                                                                                                                                                                                                                   | 1/40166 [00:45<507:45:06, 45.51s/it]
!! Mean absolute value of trainable parameters:  116.50000000
!! Mean absolute value of trainable parameters:  116.50000000
{'loss': 0.0001, 'grad_norm': 4.658809185028076, 'learning_rate': 9.99950206642434e-07, 'completion_length': 99.2890625, 'rewards/iou_reward': 0.08052092604339123, 'rewards/format_reward_rec': 0.109375, 'reward': 0.18989592790603638, 'reward_std': 0.41304440796375275, 'kl': 0.001346588134765625, 'clip_ratio': 0.0, 'epoch': 0.0}
  0%|                                                                                                                                                                                                                                   | 2/40166 [01:22<452:35:27, 40.57s/it]
!! Mean absolute value of trainable parameters:  116.50000000
!! Mean absolute value of trainable parameters:  116.50000000
{'loss': 0.0001, 'grad_norm': 2.5706093311309814, 'learning_rate': 9.999253099636507e-07, 'completion_length': 116.09375, 'rewards/iou_reward': 0.046912169083952904, 'rewards/format_reward_rec': 0.0703125, 'reward': 0.11722467094659805, 'reward_std': 0.30162496864795685, 'kl': 0.0025482177734375, 'clip_ratio': 0.0, 'epoch': 0.0}
  0%|                                                                                                                                                                                                                                   | 3/40166 [03:18<834:31:53, 74.80s/it]
!! Mean absolute value of trainable parameters:  116.50000000
!! Mean absolute value of trainable parameters:  116.50000000
{'loss': 0.0001, 'grad_norm': 2.730163812637329, 'learning_rate': 9.999004132848678e-07, 'completion_length': 99.234375, 'rewards/iou_reward': 0.07206234335899353, 'rewards/format_reward_rec': 0.1328125, 'reward': 0.20487484335899353, 'reward_std': 0.39900827407836914, 'kl': 0.002685546875, 'clip_ratio': 0.0, 'epoch': 0.0}
  0%|                                                                                                                                                                                                                                   | 4/40166 [03:50<648:14:15, 58.11s/it]
!
```

I would like to know is this normal?

**Comment by StinkyTofu95** [comment]
mark

**Comment by hanyangclarence** [comment]
But when running the training script, the loss is indeed being reduced, and the grad norm is also non-zero. So I think the training is totally fine, just that the printed parameters are not changed for some reasons..

**Comment by go-ahead-maker** [comment]
@hanyangclarence Hi, did you find that the loss **increased** at the beginning of training?  Is that normal for RL training?

**Comment by hanyangclarence** [comment]
I found that the loss first increases then decreases, while the reward is steadily increasing. I'm not sure whether this is normal cuz I'm also new to RL lol.
